src/Appwrite/Platform/Workers/Builds.php

```diff
@@ -433,6 +433,8 @@ protected function buildDeployment(Device $deviceForFunctions, Func $queueForFun
                                         throw new \Exception('Build not found', 404);
                                     }
 
+                                    $logs = \mb_substr($logs, 0, \mb_strlen($logs, 'UTF-8') - 1, 'UTF-8'); // Get only valid UTF8 part - removes leftover half-multibytes causing SQL errors
```

Comment:
what is the need for `mb_strlen` here? if you pass `null` for the length in `mb_substr` it achieves the same results